modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ForEachProcessor.java

```
@@ -78,18 +79,24 @@ String getField() {
         return field;
     }
 
-    List<Processor> getProcessors() {
-        return processors;
+    Processor getProcessor() {
+        return processor;
     }
 
     public static final class Factory implements Processor.Factory {
         @Override
         public ForEachProcessor create(Map<String, Processor.Factory> factories, String tag,
                                        Map<String, Object> config) throws Exception {
             String field = readStringProperty(TYPE, tag, config, "field");
-            List<Map<String, Map<String, Object>>> processorConfigs = readList(TYPE, tag, config, "processors");
-            List<Processor> processors = ConfigurationUtils.readProcessorConfigs(processorConfigs, factories);
-            return new ForEachProcessor(tag, field, Collections.unmodifiableList(processors));
+            Map<String, Map<String, Object>> processorConfig = readMap(TYPE, tag, config, "processor");
+            Iterator<Map.Entry<String, Map<String, Object>>> it = processorConfig.entrySet().iterator();
```

maybe throw an error if `processorConfig.entrySet().size != 1`?